/*
 * Imagery message definitions.
 *
 * Note that all units will be in meters, meters/second, seconds,
 * degrees, etc. unless otherwise noted.
 *
 * Time is in seconds from 1970 epoch.
 *
 * Yaw is in the range [0, 360) degrees and lat, lon, pitch and roll
 * are in the range (-180, 180].
 */

syntax = "proto3";

import "telemetry.proto";

package imagery;

message ImageCount {
    double time = 1;
    uint32 count = 2;
}

message AvailableImages {
    double time = 1;
    uint32 count = 2;
    repeated uint32 id_list = 3;
}

message Image {
    // Time at which the image was taken.
    double time = 1;
    uint32 id = 2;
    // True if we have telemetry for this image.
    bool has_telem = 3;
    telemetry.CameraTelem telem = 4;
    // The image is a JPEG.
    bytes image = 5;
}
